### ISSUE TITLE
PGXP Incompatibility List

### ISSUE THREAD

**Issue by stenzek** [stenzek/duckstation#733]
This issue is to keep track of games which are incompatible or broken with PGXP.

As it stands, PGXP in DuckStation is considered "finished for now", and any incompatibilities are not considered to be critical as it is an enhancement.

Implementing Beetle's "CPU mode" is not planned as it would significantly slow down the emulator, and is apparently buggy anyway. For the games which require it, there may be another solution which doesn't slow things down so much, which I might look into at some point if I have the time/motivation.

**Comment by Richard-L** [comment]
Black texture flicker is still apparent with PGXP turned on in "SLES-00024" "Tomb Raider (Europe) (No EDC)" "PAL" 
v0.1-1467-g5df116b6

(on legs)
![image](https://user-images.githubusercontent.com/4934209/89798435-79661a80-db2c-11ea-8b01-009af441c7b9.png)
![image](https://user-images.githubusercontent.com/4934209/89798551-97cc1600-db2c-11ea-993c-5b33a46dae67.png)

**Comment by stenzek** [comment]
The Tomb Raider issue seems to be specific to older hardware? I can't reproduce it on either Intel (Kaby Lake) or NVIDIA (Pascal) GPUs.

**Comment by RinMaru** [comment]
Happens to me aswell AMD RX 480

**Comment by RinMaru** [comment]
**Games Affected By PGXP**

1. Crash Bandicoot Warped (**SCUS-94244**) - _Visible seams in geometry from a distance_
2. Tomb Raider (**SLUS-00152**) - _Corrupted textures on geometry_

**Comment by stenzek** [comment]
> Crash Bandicoot Warped (USA) (SCUS-94244) - Visible black blocks in geometry from a distance

This likely isn't a PGXP bug - the game uses precomputed culling data, so it's probably unfixable. Unless it's fine in other emulators.

> Tomb Raider (USA) (SLUS-00152) - Corrupted textures on geometry
> Tomb Raider (PAL) (SLES-00024) - Corrupted textures on geometry

Still can't reproduce any of these. I'll try on an AMD GPU tomorrow.

**Comment by RinMaru** [comment]
You can easily see issues with the headphone's on the start screen. I dont recall BeetleHW having the issue on Crash 3 my memory is bad so dont take my word for it

**Comment by bigol83** [comment]
Less visible but still happening 

![Gran Turismo 2 (Europe) (Disc 2) (Gran Turismo Mode Disc)-200810-205525](https://user-images.githubusercontent.com/38129260/89819622-08365f80-db4c-11ea-914b-4b8e52ddc6d9.png)


https://github.com/stenzek/duckstation/issues/685

Gran Turismo 2 PAL (SCES-12380) White lines on wheels with PGXP enabled

**Comment by Pik-a-boo** [comment]
PGXP wont work with games that got developed by Argonaut (excluding Alien: Resurrection) and Alundra 2.

**Comment by Blackbird88** [comment]
Metal Gear Solid still has flickering water but this time instead of being fully corrupted it's some weird color
![image](https://user-images.githubusercontent.com/4414625/89869792-2a17fc80-dbb5-11ea-905c-9606d5041d1e.png)

**Comment by saintbepis** [comment]
Front mission 3 has flickering title graphics with PGXP

**Comment by stenzek** [comment]
Please re-test these with the latest build (https://github.com/stenzek/duckstation/commit/e6156e2adbf8c00b18b1aebf8f5ada036cd68c1d). Tomb Raider is fixed for me now (headphones on menu screen).

**Comment by RinMaru** [comment]
its Fixed

**Comment by Richard-L** [comment]
Also fixed for me 👍 

What was the problem?

**Comment by bigol83** [comment]
Issue with white lines in GT2 is still there for me, even if less visible. Noticeable on anterior wheel

![Gran Turismo 2 (Europe) (Disc 2) (Gran Turismo Mode Disc)-200811-205038](https://user-images.githubusercontent.com/38129260/89936981-bdcce580-dc14-11ea-8552-e58ab3fb3bef.png)

I also noticed that vertex cache improves this issue but introduces bugs in menus like this one

Vertex cache ON 
A lot less visible white lines on wheels, but lines on menu elements 

![Gran Turismo 2 (Europe) (Disc 2) (Gran Turismo Mode Disc)-200811-205841](https://user-images.githubusercontent.com/38129260/89937696-bd811a00-dc15-11ea-993e-2b0e6c2a27ec.png)

Vertex cache OFF
A lot more white lines but correct menu elements

![Gran Turismo 2 (Europe) (Disc 2) (Gran Turismo Mode Disc)-200811-205853](https://user-images.githubusercontent.com/38129260/89937689-bce88380-dc15-11ea-9ba0-cb6e935e3a85.png)

**Comment by Blackbird88** [comment]
MGS1 is fixed. Thanks!

**Comment by bajolzas** [comment]
Grandia is also not fixed with memory PGXP.
On Vandal Hearts II's initial cutscene, whenever the screen moves, character sprites become unstable.
both are fixed when using CPU PGXP, on PCSXR.

**Comment by HeroponRikiBestest** [comment]
Vertex Cache breaks the UI in Ganbare Goemon- Kuru nara Koi! Ayashige Ikka no Kuroi Kage (Japan) in Vulkan and OpenGL (I can't test DirectX, but I'd imagine it's broken there as well)
Vertex Cache on:
![image](https://user-images.githubusercontent.com/50224630/90324946-19a3b100-df2a-11ea-8873-fe626389e15a.png)
Vertex Cache off:
![image](https://user-images.githubusercontent.com/50224630/90324950-21fbec00-df2a-11ea-90bd-39fc155d274a.png)

**Comment by IPeluchito** [comment]
Silent Hill options with **PGXP** enabled and **(2x-16x)** scaling, at **1x** it works normally ...

![SLUS-00707_2020-08-18_14-17-37](https://user-images.githubusercontent.com/64821708/90558799-4cd08500-e162-11ea-8aa2-36b3db5db2f7.png)

**Comment by stenzek** [comment]
> Silent Hill options with **PGXP** enabled and **(2x-16x)** scaling, at **1x** it works normally ...
> 
Silent Hill seems fine here with PGXP and upscaling on.
![SLUS-00707_2020-08-21_15-01-30](https://user-images.githubusercontent.com/11288319/90854063-425ef700-e3bf-11ea-8dc4-0a16d75180d4.png)

**Comment by IPeluchito** [comment]
I just tried it again ... in Silent Hill once the game has started, enter options and if you have Vertex Cache enabled, the 2D textures of options are distorted ... they happen with several games, it is not the only one, in Resident Evil Director's Cut in-game subtitles shake ... if Vertex cache is disabled there is no distortion or shaking ...

Windows 8.1, Vulkan Render, 12x, All video enhancements except Integer Upscaling, Bilinear Texture Filtering and CPU Mode (pgxp)

**Comment by MojoJojoDojo** [comment]
> PGXP wont work with games that got developed by Argonaut (excluding Alien: Resurrection) and Alundra 2.

fixed with PGXP CPU mode.

**Comment by HeroponRikiBestest** [comment]
PGXP just doesn't do anything in any of the tony hawk games. It also introduces some extra issues in thps4, which are identical to the beetle hw issues seen in https://github.com/libretro/beetle-psx-libretro/issues/576.

**Comment by IPeluchito** [comment]
**SLUS-00674 International Superstar Soccer Pro 98 (VERTEX CACHE=ON)** After a few matches the graphics get completely corrupted ...

![SLUS-00674_2020-09-05_21-43-41](https://user-images.githubusercontent.com/64821708/92317147-367c5300-efc3-11ea-99ee-abce9d490310.png)

Windows 8.1, Vulkan Render, 12x, All video enhancements except Integer Upscaling, Bilinear Texture Filtering and CPU Mode (pgxp)

**Comment by stenzek** [comment]
You shouldn't be using vertex cache. I haven't found any games where it actually improves things, so I'm probably going to move it to the advanced menu.

**Comment by stenzek** [comment]
Gran Turismo 1/2 polygon holes should be fixed.

**Comment by DonelBueno** [comment]
There are what appears to be culling problems in CTR with PGXP on (these are not present in RA), no PGXP setting changes it:

![DuckStation_2020_10_04_15_32_30_137](https://user-images.githubusercontent.com/4239694/95024228-17232500-0658-11eb-8abe-dfa510576b49.png)
![DuckStation_2020_10_04_15_32_34_293](https://user-images.githubusercontent.com/4239694/95024230-18545200-0658-11eb-9661-d6ee01babdab.png)

Spyro trilogy has some serious issues with PGXP on (again, these don't happen in RA):

![DuckStation_2020_10_04_15_34_41_718](https://user-images.githubusercontent.com/4239694/95024232-19857f00-0658-11eb-80f2-2f5c2b5a61a6.png)
![DuckStation_2020_10_04_15_34_47_492](https://user-images.githubusercontent.com/4239694/95024235-1a1e1580-0658-11eb-85e4-68a85cb28bbf.png)

**Comment by stenzek** [comment]
> There are what appears to be culling problems in CTR with PGXP on (these are not present in Beetle HW), no PGXP setting changes it:
Also broken in beetle psx. Fixed with vertex cache.

> Spyro trilogy has some serious issues with PGXP on (again, these don't happen in Beetle HW):

DS behaves the same as beetle here, except for the exploding triangles with vertex cache enabled. I'll look into it.

In the future, please don't make false claims and test with equivalent settings. It pains me to use retroarch because I always do something wrong (e.g. this time upscaling is refusing to work) and have to mess around with it just to verify a claim which isn't even true.

**Comment by stenzek** [comment]
After checking Spyro again, it's CPU mode which breaks it not vertex cache, and you should have CPU mode off globally anyway. Therefore I don't think either of these issues are valid.

**Comment by DonelBueno** [comment]
Yes, I have CPU mode disabled globally, but enabled for Spyro trilogy. The issue is valid, I just forgot to mention it happened only in CPU mode (memory only mode is basically useless in this trilogy).

**Comment by stenzek** [comment]
Spyro issue fixed in https://github.com/stenzek/duckstation/commit/fa638ef9d2bfbfde7ac628797f0e51d810502841

> As for CTR, Vertex Cache doesn't fix it for me, and Beetle HW in RetroArch doesn't need vertex cache to to get rid of the glitches.

It was still broken in Beetle for me, maybe I had some setting wrong because like I said upscaling was broken too, but I'm not spending any more time with that slow, broken junk.

**Comment by DonelBueno** [comment]
Thanks for the quick fix.

Is it possible to make PGXP CPU mode compatible with the recompiler someday?

**Comment by stenzek** [comment]
CTR issue is actually due to increased precision vs Beetle. I'll probably default it to off and make it an option, it does look ever-so-slightly better in some games like Tomb Raider.

> Is it possible to make PGXP CPU mode compatible with the recompiler someday?

No. Doing so would defeat the point of the recompiler (calling out to C code for every instruction), and inlining it would balloon code size massively, significantly reducing any performance gain from the recompiler in the first place.

I already use code caching/the cached interpreter for CPU mode, so it's not like you're forced back to the pure interpreter. It's only needed for a handful of games anyway.

**Comment by HeroponRikiBestest** [comment]
All 3 PGXP options have no effect on Courier Crisis - The Saga of the Modern Fatalist (USA)

**Comment by MojoJojoDojo** [comment]
> All 3 PGXP options have no effect on Courier Crisis - The Saga of the Modern Fatalist (USA)

It could be that it needs PGXP CPU but its only present on per a game settings.

**Comment by angrylion7** [comment]
Enabling "texture correction" causes the skybox to flicker in Test Drive Off-Road 2 US version (SLUS-00609). The fastest way to see is choose Single Race, then Hawaii track. Couldn't reproduce this in PCSXR-PGXP and Beetle-PSX-HW. Tested with D3D11 and OpenGL renderers on an nVidia Kepler GPU.

**Comment by Kurayami7** [comment]
PGXP doesn't work during the opening title screen of Covert Ops - Nuclear Dawn (USA). Tested with CPU mode on and off. Confirmed to not be an FMV.

![SLUS-01151_2021-01-21_02-56-00](https://user-images.githubusercontent.com/77198254/105211768-c270cf00-5ba0-11eb-9690-374923925b1a.png)

Text also breaks with CPU mode.

![SLUS-01151_2021-01-21_03-03-38](https://user-images.githubusercontent.com/77198254/105211792-c997dd00-5ba0-11eb-9135-502a967b644c.png)

**Comment by Kurayami7** [comment]
Graphics in Kickboxing (USA) break with PGXP.

![SLUS-01412_2021-01-22_05-55-44](https://user-images.githubusercontent.com/77198254/105398587-a3dd0780-5c76-11eb-8b5e-eb877af197b8.png)

**Comment by a-hurst** [comment]
The Soliton radar in Metal Gear Solid (Disc 1) (USA) becomes progressively more broken with time when PGXP is turned on. The visual fields of your opponents + cameras (the yellow cones) display fine at first (e.g. if I turn PGXP off and then back on), but if I move around a bit the visual fields become increasingly detached from their corresponding red dots and will start to jump around on the HUD relative to my movement, sometimes disappearing entirely. The cones will also sometimes glitch for a second and point in a completely different direction for a frame or two before returning to normal.

Slightly detached:

![SLUS-00594_2021-01-25_18-47-31](https://user-images.githubusercontent.com/18648066/105778790-7082cd00-5f43-11eb-8e9f-60b58de72580.png)

Very detached, after moving around for ~20 seconds:

![SLUS-00594_2021-01-25_19-17-28](https://user-images.githubusercontent.com/18648066/105778824-82fd0680-5f43-11eb-82e0-d2a271fcbe94.png)

The issue disappears immediately when I turn PGXP off. I've tried PGXP alone, PGXP + Vertex Cache, PGXP + CPU, PGXP with increased precision, basically every combination I can think of, but nothing seems to make a difference. Based on the "getting worse over time" thing, maybe it's related to the invalidation issue mentioned at the end of [this thread](https://github.com/stenzek/duckstation/issues/1053)?

This also doesn't seem to be bug specific to my hardware, since it likewise shows up in a DuckStation play-test of the game I found on YouTube (see [17:10 to 17:30](https://youtu.be/hoVn1jwA1wI?t=1032) for a clear illustration of the issue in action) and I doubt they're running the game at 1440p on an HD3000.

**Comment by backgamon** [comment]
PGXP change nothing (jitter + unprecision) with Eutechnyx game engine:

M6 Turbo Racing/Total Driving/Grand Tour Racing 98
C3 Racing/Max Power Racing
Le Mans 24H
007 Racing
TNN Hardcore TR

And crash after a short while with:
Lego Island 2: The Brickster's Revenge/L'Île Lego 2 : La Revanche de Casbrick
-> Both Memory Only/CPU.

**Comment by waspennator** [comment]
PGXP Culling Correction needs to be turned off for Family Feud (USA), cause it freezes as soon as you pick a family and start the game.

**Comment by RaduNastase** [comment]
Rival Schools [USA] issue:

![Screenshot_20210426-075209](https://user-images.githubusercontent.com/18061144/116037143-59ea3000-a670-11eb-8529-0d65418605de.jpg)

This happends in a specific rotation zone (sides step) of each level, when you launch a character in the air (press forward+down + P).

In some other rotation zones, the fence dissappears:

![Screenshot_20210426-075248](https://user-images.githubusercontent.com/18061144/116037469-d1b85a80-a670-11eb-8a6e-d3d2266bb425.jpg)


Happends in both standalone and Retroarch Core.

**Comment by JOHN30011887** [comment]
Spyro 1, 2 and 3 dont work properly when pgxp is on, no matter the settings in that section toggled off or on, you see through the game world at the top edge of the screen, its like small triangles popping in and out when panning camera

Can be seen best when indoors/in caves etc

**Comment by stenzek** [comment]
> Spyro 1, 2 and 3 dont work properly when pgxp is on, no matter the settings in that section toggled off or on, you see through the game world at the top edge of the screen, its like small triangles popping in and out when panning camera
> 
> Can be seen best when indoors/in caves etc

That's because Spyro's engine messes with geometry after transforming it. You can use CPU mode to work around this, but it's still not perfect - would need a game patch to remove the vertex postprocessing.

**Comment by waspennator** [comment]
NCAA Final Four 2000 (USA) needs CPU mode on by default when enabling PGXP or it just weirdly breaks.

Without CPU mode:
![SCUS-94562_2021-06-30_11-54-17](https://user-images.githubusercontent.com/78768499/124009402-cb10d100-d99a-11eb-94ec-08329de1739a.png)



With CPU mode:
![SCUS-94562_2021-06-30_12-06-45](https://user-images.githubusercontent.com/78768499/124010266-d9abb800-d99b-11eb-9b19-33f26a9e74e6.png)

**Comment by Luke3357** [comment]
Resident Evil 3 with PGXP and CPU Mode On makes the flames invisible
![SLUS-00923_2021-07-18_01-22-41](https://user-images.githubusercontent.com/12575618/126055775-93d10c6a-66bf-4fad-9fdc-c7ed9daf8268.png)

**Comment by Awakened0** [comment]
I found a freeze that happens in a specific cutscene of Klonoa if PGXP Geometry Correction is enabled (using win64 build 0.1-4599-ga542fa85, dx11 renderer):

https://user-images.githubusercontent.com/4565863/128428762-085ce60f-4e87-4ab5-b6ff-374545bcbe48.mp4

Here's a log: https://pastebin.com/jMWNaxA0

You have to have a specific save before the stage and boss to play this cutscene. I ran through the game a second time to get a proper save file:

[Klonoa - Door to Phantomile (USA)_1.zip](https://github.com/stenzek/duckstation/files/6941972/Klonoa.-.Door.to.Phantomile.USA._1.zip)

After loading that save, you'll have to play through the level and beat the boss to get to the cutscene that freezes.

**Comment by F3arM3** [comment]
I hope this is the right place, geometry correction in tomorrow never dies creates cracks in character models unless in cpu mode

Without CPU mode
![SLUS-00975_2021-08-18_19-09-02](https://user-images.githubusercontent.com/89053368/129987679-15e31aeb-2732-4ad2-84da-c2b342bec944.png)


With CPU mode
![SLUS-00975_2021-08-18_19-06-12](https://user-images.githubusercontent.com/89053368/129987638-f1d9b08a-c147-4040-b5aa-19108ef1ba3c.png)

**Comment by catscandal** [comment]
PGXP seems to cause a softlock in some Silent Hill cutscenes. Game ran perfectly with Geometry Correction, Texture Correction, and Culling Correction on for the first few hours, but softlocked at the cutscene after the first time entering the Balkan Church. Turning off those options fixed it.

Only other enhancements at non-recommended values were 5x resolution scale and disable interlacing.

**Comment by stenzek** [comment]
> PGXP seems to cause a softlock in some Silent Hill cutscenes

Yes, this is known. It's because the game uses fixed size geometry buffers, and the extra geometry not culled because of PGXP causes said buffer to overflow. Same happens with the WS hack. Nothing an emu can do without seriously modding the game. Turn off culling correction for that part, then re-enable it.
